chicken-sloths/bangazon-site-sprint3
# Add Products to Sell

# Description
This PR goes in tandem with @melissabell456's PR to manage products. On her PR (which will be on the route `products/manage`, there'll be a button with a hyperlink to add products. This PR takes care of the form to add a new product. Once @DavidLarsKetch is done with the prod details view, this form will redirect to the prod details view on success.

## Related Ticket(s)
Fixes #33 , fixes #28, fixes #15, fixes #40 

## Problem to Solve
Creates a form for users to post a new product.

## Proposed Changes
1. I broke out a `routes/productsRouter.js` and required it into `routes/index.js`. I know we haven't all been doing this but it made sense for mine to go ahead and break out. I coordinated with Melissa to make sure our routes work together.

## Expected Behavior & Steps to Test
1. Run `npm run db:gen`
1. Run `npm start`
1. Go to `/products/manage/new`
1. Fill out the form!
1. You should see a json response of the product you just added. (Eventually this will re-route to DLK's product details page, the JSON is a placeholder to confirm that it works)
1. If you go into your database, either in the psql terminal or in your db browser, you should see the product you just added.


## Notes
A couple things:
- It looks like I deleted `app/controllers/config/config.json`. Here's what really happened: I accidentally added it because I thought it was gitignored. (and apparently it was gitignored? I'm still not sure what happened.) ANYWAY all I did was tell git to untrack it with `git -rm -r <filename>` and now git is like `woah woah woah you deleted it!!` but it's still there on my local system and it should still be there on your local system. If you pull this branch down and don't see `config.json` where it should be, let me know and I'll fix it.
- Right now we can't get the current user id because joe's boilerplate code doesn't work the way we thought it did. He's fixing it. All new products are being posted with a user if of `1` for the moment.
